### PR TITLE
chore(server): Introduce StringSetWrapper

### PR DESCRIPTION
**Pull request by dranikpg**
Weekend cleanup after I closed my pending tasks 🛀🏻

### CODE REVIEW

File: src/server/set_family.cc [hunk 1]
```diff
@@ -74,58 +74,74 @@ intset* IntsetAddSafe(string_view val, intset* is, bool* success, bool* added) {
...
 
-    for (string_view member : vals) {
-      removed += ss->Erase(member);
-    }
+  static void Init(CompactObj* obj) {
+    obj->InitRobj(OBJ_SET, kEncodingStrMap2, CompactObj::AllocateMR<StringSet>());
+  }
 
-    isempty = ss->Empty();
+  unsigned Add(const NewEntries& entries, uint32_t ttl_sec) const {
```
Comment (kostasrim) [inline src/server/set_family.cc:92]: Why do we use unsigned an not `uint32_t or uint64_t` ? It's less verbose :rofl:

File: src/server/set_family.cc [hunk 2]
```diff
@@ -74,58 +74,74 @@ intset* IntsetAddSafe(string_view val, intset* is, bool* success, bool* added) {
...
-}
+ private:
+  StringSetWrapper(void* robj_ptr, uint64_t now_ms) : ss(static_cast<StringSet*>(robj_ptr)) {
+    ss->set_time(MemberTimeSeconds(now_ms));
+  }
 
-void InitStrSet(CompactObj* set) {
-  set->InitRobj(OBJ_SET, kEncodingStrMap2, CompactObj::AllocateMR<StringSet>());
-}
+  StringSet* const ss;
```
Comment (kostasrim) [inline src/server/set_family.cc:134]: west side vs east side :eyes:

File: src/server/set_family.cc [hunk 3]
```diff
@@ -74,58 +74,74 @@ intset* IntsetAddSafe(string_view val, intset* is, bool* success, bool* added) {
...
+  auto Range() const {
+    auto transform = [](const sds ptr) { return string_view{ptr, sdslen(ptr)}; };
+    return base::it::Transform(transform, base::it::Range(ss->begin(), ss->end()));
   }
 
-  return res;
-}
+ private:
+  StringSetWrapper(void* robj_ptr, uint64_t now_ms) : ss(static_cast<StringSet*>(robj_ptr)) {
+    ss->set_time(MemberTimeSeconds(now_ms));
```
Comment (kostasrim) [inline src/server/set_family.cc:131]: Just make sure that we don't call this on paths that don't require this. I went over the cases and they seemed fine but just double checking

**Review by kostasrim — APPROVED** [review]

File: src/server/set_family.cc (hunk 3 above)
Comment (dranikpg) [reply src/server/set_family.cc:131]: I checked too. And if a command path existed, it was likely missing, not abundant 🤷🏻‍♂️

**Pull request merged**